src/Common/src/TypeSystem/RuntimeDetermined/RuntimeDeterminedType.cs

```diff
@@ -127,7 +127,13 @@ protected override TypeFlags ComputeTypeFlags(TypeFlags mask)
 
         public override TypeDesc GetTypeDefinition()
         {
-            return _rawCanonType.GetTypeDefinition();
+            // TODO: this is needed because NameMangler calls it to see if we're dealing with genericness. Revise?
```

Comment:
Targets of ready to run helpers are runtime determined. I could probably introduce a separate name mangler for those.

But the semantics of this method is still iffy if I leave it as is.